packages/amplify_auth_cognito/lib/src/CognitoDevice/cognito_device.dart

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:amplify_auth_cognito/amplify_auth_cognito.dart';
+import 'package:collection/collection.dart';
+
+/// Helper functions for [AuthDevice].
+extension AuthDeviceX on AuthDevice {
```

Comment:
This is interesting. I have wanted to put some time into how we can improve the DX for some of the APIs that have Cognito specific implementations. I had not thought of using extensions to at least make casting it easier. `AuthSession`/`CognitoAuthSession` is another place this would be useful. Probably a bunch more.

I still think there might be a larger change we can make to improve the DX of the Cognito specific APIs but this is definitely an improvement 👍

Reply:
Word 👍